client/functions.lua

```diff
@@ -151,45 +181,45 @@ local function lockpickCallback(vehicle, plate, isAdvancedLockedpick, maxDistanc
 --- @param isAdvancedLockedpick boolean Determines whether an advanced lockpick was used
-function public.lockpickDoor(isAdvancedLockedpick)
-    local maxDistance = 2
+--- @param maxDistance number? The max distance to check.
+--- @param customChallenge boolean? lockpick challenge
+function public.lockpickDoor(isAdvancedLockedpick, maxDistance, customChallenge)
+    maxDistance = maxDistance or 2
     local pedCoords = GetEntityCoords(cache.ped)
     local vehicle = lib.getClosestVehicle(pedCoords, 4, false)
 
     if not vehicle then return end
 
     local plate = qbx.getVehiclePlate(vehicle)
     local isDriverSeatFree = IsVehicleSeatFree(vehicle, -1)
-    local doorCoords = getBoneCoords(vehicle, 'door_dside_f')
 
     --- player may attempt to open the lock if:
-    if not vehicle
-        or not plate
-        or not isDriverSeatFree                                               -- no one in the driver's seat
-        or Entity(vehicle).state.isOpen                                       -- the lock is locked
-        or not isCloseToCoords(doorCoords, pedCoords, maxDistance)            -- the player's ped is close enough to the driver's door
-        or GetVehicleDoorLockStatus(vehicle) < 2                              -- the vehicle is locked
-        or lib.callback.await('qbx_vehiclekeys:server:hasKeys', false, plate) -- player does not have keys to the vehicle
-    then
-        return
-    end
-
-    lockpickingSemaphore = lockpickingSemaphore + 1 -- semaphore
+    if not plate
+        or not isDriverSeatFree                                             -- no one in the driver's seat
+        or public.hasKeys(plate)                                            -- player does not have keys to the vehicle
+        or Entity(vehicle).state.isOpen                                     -- the lock is locked
+        or not isCloseToAnyBone(pedCoords, vehicle, doorBones, maxDistance) -- the player's ped is close enough to the driver's door
+        or GetVehicleDoorLockStatus(vehicle) < 2                            -- the vehicle is locked
+    then return end
+
+    lockpickingSemaphore = lockpickingSemaphore + 1
     if lockpickingSemaphore > 1 then return end
-    Wait(0)
 
     CreateThread(function()
         --- lock opening animation
         lib.requestAnimDict('veh@break_in@0h@p_m_one@')
         TaskPlayAnim(cache.ped, 'veh@break_in@0h@p_m_one@', "low_force_entry_ds", 3.0, 3.0, -1, 16, 0, false, false, false)
 
-        local isSuccess = lib.skillCheck({ 'easy', 'easy', { areaSize = 60, speedMultiplier = 1 }, 'medium' },
-            { '1', '2', '3', '4' })
+        local isSuccess = customChallenge or
+            lib.skillCheck({ 'easy', 'easy', { areaSize = 60, speedMultiplier = 1 }, 'medium' },
+                { '1', '2', '3', '4' })
 
         lockpickCallback(vehicle, plate, isAdvancedLockedpick, maxDistance, isSuccess)
+
+        Wait(config.lockpickCooldown)
     end)
 
     lockpickingSemaphore = 0
```

Comment:
This needs to be inside the thread, no?

Reply:
meaningless, the thread is just to skillCheck run smoothly

Reply:
If it's not locking the skill check, then one could create multiple skill check threads at the same time, no?